src/map/clif.c

```diff
@@ -20672,6 +20676,192 @@ void clif_private_airship_response(struct map_session_data *sd, uint32 flag)
+	struct stylist_data_entry *entry;
+
+	nullpo_retr(false, sd);
+	Assert_retr(false, type >= 0 && type < MAX_STYLIST_TYPE);
+	Assert_retr(false, idx >= 0 && idx < VECTOR_LENGTH(stylist_data[type]));
+
+	entry = &VECTOR_INDEX(stylist_data[type], idx);
+
+	if (entry->id != 0) {
+		if (entry->zeny != 0) {
+			if (sd->status.zeny < entry->zeny)
+				return false;
+
+			sd->status.zeny -= entry->zeny;
+			clif->updatestatus(sd, SP_ZENY);
+		} else if (entry->itemid != 0) {
+			it.nameid = entry->itemid;
+			it.amount = 1;
+			return script->buildin_delitem_search(sd, &it, false);
+		} else if (entry->boxid != 0) {
+			it.nameid = entry->boxid;
+			it.amount = 1;
+			return script->buildin_delitem_search(sd, &it, false);
+		}
+		return true;
+	}
+	return false;
+}
+void clif_stylist_send_rodexitem(struct map_session_data *sd, int16 itemid)
+{
+	struct rodex_message msg = { 0 };
+
+	nullpo_retv(sd);
+
+	msg.receiver_id = sd->status.char_id;
+	msg.items[0].item.nameid = itemid;
+	msg.items[0].item.amount = 1;
+	msg.items[0].item.identify = 1;
+	msg.type = MAIL_TYPE_NPC | MAIL_TYPE_ITEM;
+
+	safestrncpy(msg.sender_name, msg_txt(366), NAME_LENGTH);
+	safestrncpy(msg.title, msg_txt(367), RODEX_TITLE_LENGTH);
+	safestrncpy(msg.body, msg_txt(368), MAIL_BODY_LENGTH);
+	msg.send_date = (int)time(NULL);
+	msg.expire_date = (int)time(NULL) + RODEX_EXPIRE;
+
+	intif->rodex_sendmail(&msg);
+}
+
+void clif_parse_cz_req_style_change(int fd, struct map_session_data *sd) __attribute__((nonnull(2)));
+void clif_parse_cz_req_style_change(int fd, struct map_session_data *sd)
+{
+	const struct PACKET_CZ_REQ_STYLE_CHANGE *p = RP2PTR(fd);
+
+	if (p->HeadStyle > 0)
+		clif->cz_req_style_change_sub(sd, LOOK_HAIR, p->HeadStyle, false);
+	if (p->HeadPalette > 0)
+		clif->cz_req_style_change_sub(sd, LOOK_HAIR_COLOR, p->HeadPalette, false);
+	if (p->BodyPalette > 0)
+		clif->cz_req_style_change_sub(sd, LOOK_CLOTHES_COLOR, p->BodyPalette, false);
+	if (p->TopAccessory > 0)
+		clif->cz_req_style_change_sub(sd, LOOK_HEAD_TOP, p->TopAccessory, true);
+	if (p->MidAccessory > 0)
+		clif->cz_req_style_change_sub(sd, LOOK_HEAD_MID, p->MidAccessory, true);
+	if (p->BottomAccessory > 0)
+		clif->cz_req_style_change_sub(sd, LOOK_HEAD_BOTTOM, p->BottomAccessory, true);
+
+	clif->style_change_response(sd, true);
+	return;
+}
+
+void clif_cz_req_style_change_sub(struct map_session_data *sd, int type, int16 idx, bool isitem)
+{
+	struct stylist_data_entry *entry;
+
+	nullpo_retv(sd);
+	Assert_retv(idx > 0);
+	Assert_retv(type >= 0 && type < MAX_STYLIST_TYPE);
+
+	if ((idx - 1) < VECTOR_LENGTH(stylist_data[type])) {
+		entry = &VECTOR_INDEX(stylist_data[type], idx - 1);
+		if (clif->style_change_validate_requirements(sd, type, idx - 1)) {
+			if (isitem == false)
+				pc->changelook(sd, type, entry->id);
+			else
+				clif->stylist_send_rodexitem(sd, entry->id);
+		}
+	}
+}
+
+void clif_style_change_response(struct map_session_data *sd, int8 flag)
+{
+	struct PACKET_ZC_STYLE_CHANGE_RES p;
```

Comment:
here need add limit for send packet to only supported clients
``#if PACKETVER >= 20151104``